compiler/damlc/daml-ide/src/DA/Daml/LanguageServer/Visualize.hs

```diff
@@ -0,0 +1,56 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- | Execute Daml commands
+
+module DA.Daml.LanguageServer.Visualize
+    ( setCommandHandler
+    ) where
+
+import qualified Data.Aeson as Aeson
+import           Language.Haskell.LSP.Types
+import           Development.IDE.LSP.Server
+
+import Development.IDE.Types.Logger
+
+import qualified Data.Text as T
+
+import Development.IDE.Core.Rules
+import Development.IDE.Core.Service.Daml
+
+import Development.IDE.Core.Shake
+import Development.IDE.Core.RuleTypes.Daml
+
+import Language.Haskell.LSP.Messages
+import qualified Language.Haskell.LSP.Core as LSP
+import Development.IDE.Types.Location
+
+collectTexts :: List Aeson.Value -> Maybe NormalizedFilePath
+collectTexts (List [Aeson.String file])  = Just (toNormalizedFilePath (T.unpack file))
+collectTexts _= Nothing
+
+onCommand
+    :: IdeState
+    -> ExecuteCommandParams
+    -> IO Aeson.Value
+onCommand ide execParsms = case execParsms of
+    ExecuteCommandParams "daml/damlVisualize" (Just _arguments) -> do
+        case collectTexts _arguments of
+            Just mod -> do
+                    logInfo (ideLogger ide) "Generating visualization for current daml project"
+                    Just dots <- runAction ide (use GenerateVisualization mod)
+                    return $ Aeson.String dots
+            Nothing     -> do
+                logError (ideLogger ide) "Expected a single module to visualize, got multiple module"
+                return $ Aeson.String "Expected a single module to visualize, got multiple module"
```

Comment:
This is still not going via the error mechanism in LSP. It looks like doing that will require modifications to `withResponse` so I’m fine with doing that in a separate PR but please add a `TODO`.

Reply:
sounds good. Done